main/src/cgeo/geocaching/MainActivity.java

```diff
@@ -101,56 +102,62 @@
     private ConnectivityChangeReceiver connectivityChangeReceiver;
 
     private final UpdateLocation locationUpdater = new UpdateLocation();
+    private final Handler updateUserInfoHandler = new UpdateUserInfoHandler(this);
+    private final Handler firstLoginHandler = new FirstLoginHandler(this);
 
-    private final Handler updateUserInfoHandler = new Handler() {
+    private static final class UpdateUserInfoHandler extends Handler {
+        private final WeakReference<MainActivity> activityRef;
```

Comment:
That's not possible, because `MainActivity` does not extend `Activity`, which is expected by `WeakReferenceHandler`.